src/routes/settings.tsx

```diff
@@ -14,10 +13,23 @@ export const Route = createFileRoute("/settings")({
 });
 
 function Settings() {
-  const { booksPerSearch, setBooksPerSearch, language, setLanguage, backendURL, setBackendURL } = useSettingsStore();
+  const { booksPerSearch, setBooksPerSearch, language, setLanguage, backendURL } = useSettingsStore();
+  const [error, setError] = React.useState<string | null>(null);
+
+  const handleBooksPerSearchChange = (e: React.ChangeEvent<HTMLInputElement>) => {
```

Comment:
This will get annoying to do for more fields fast. Which is why this entire file should be refactored to use this

https://ui.shadcn.com/docs/components/form